tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -2449,24 +2449,6 @@ void CGMSHLSLRuntime::AddHLSLFunctionInfo(Function *F, const FunctionDecl *FD) {
     }
   }
 
-  // If InputNodes is empty, add an implicit input.
```

Comment:
We need to chek whether `SV_DispatchGrid` is necessary and fail if no input with that semantic is provided.  It's necessary when `funcProps->Node.MaxDispatchGrid[0] > 0` (`[NodeMaxDispatchGrid(...)]` was used).